core/src/main/java/jenkins/install/UpgradeWizard.java

```diff
@@ -41,13 +42,43 @@
      */
     private static final String SHOW_UPGRADE_WIZARD_FLAG = UpgradeWizard.class.getName() + ".show";
 
-    public UpgradeWizard() throws IOException {
+    public UpgradeWizard() {
         super("UPGRADE", false);
     }
     
     @Override
     public void initializeState() {
+        // Initializing this state is directly related to 
+        // running the detached plugin checks, these should be consolidated somehow
         updateUpToDate();
+        
+        // If there are no platform updates, proceed to running
+        if (isUpToDate) {
+            if (Jenkins.getInstance().getSetupWizard().getPlatformPluginUpdates().isEmpty()) {
+                Jenkins.getInstance().setInstallState(InstallState.RUNNING);
```

Comment:
in InstallUtil there is states.put(InstallState.UPGRADE, InstallState.INITIAL_SETUP_COMPLETED);

why RUNNING is forced here?